### PR TITLE
Destination bigquery: pull in cdk update for refreshes bugfix

### PR DESCRIPTION
**Pull request by edgao**
mostly just updates to make it compile against latest CDK. The big thing is adding the ability to fetch the generation ID of the final table in BQDestinationHandler.

### CODE REVIEW

**Comment by edgao** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/42504?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#42504** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/42504?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#42503** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/42503?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @edgao and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-integrations/connectors/destination-bigquery/src/main/kotlin/io/airbyte/integrations/destination/bigquery/BigQueryDestination.kt [hunk 1]
```diff
@@ -558,6 +560,7 @@ class BigQueryDestination : BaseConnector(), Destination {
 }
 
 fun main(args: Array<String>) {
+    log.info { "dumb change to force CI to actually run" }
```
Comment (gisripa) [inline airbyte-integrations/connectors/destination-bigquery/src/main/kotlin/io/airbyte/integrations/destination/bigquery/BigQueryDestination.kt:563]: lol don't forget to delete.

Comment (edgao) [reply airbyte-integrations/connectors/destination-bigquery/src/main/kotlin/io/airbyte/integrations/destination/bigquery/BigQueryDestination.kt]: derp. done, thanks for catching

**Review by gisripa — APPROVED** [review]

**Comment by edgao** [comment]
realized that (a) bigquery doesn't have coverage on generation ID migration on an overwrite stream, and (b) there's probably a bug in how we handle that migration (discovered during snowflake+redshift implementation)

copying over redshift's overwrite test case + fixing the bug.

**Pull request merged**